lodash/index.d.ts

```diff
@@ -12701,7 +12701,7 @@ declare namespace _ {
          * @param value The value to check.
          * @returns Returns true if value is correctly classified, else false.
          */
-        isWeakMap<K, V>(value?: any): value is WeakMap<K, V>;
+        isWeakMap<K, V>(value?: any): value is WeakMapLike<K, V>;
```

Comment:
this change disables the guard.  `WeakMapLike` does not even have the declarations of `WeakMap`, so it is not usable.

I would say just remove `WeakMap` declaration altogether, and make this function return boolean.